cmd/kubeapps-apis/plugins/kapp_controller/packages/v1alpha1/server_data_adapters.go

```diff
@@ -134,3 +136,51 @@ func (s *Server) getAvailablePackageDetail(pkgMetadata *datapackagingv1alpha1.Pa
 
 	return availablePackageDetail, nil
 }
+
+func (s *Server) getInstalledPackageSummary(pkgInstall *packagingv1alpha1.PackageInstall, pkgMetadata *datapackagingv1alpha1.PackageMetadata, pkgVersionsMap map[string][]pkgSemver, cluster string) (*corev1.InstalledPackageSummary, error) {
+	// get the versions associated with the package
+	versions := pkgVersionsMap[pkgInstall.Spec.PackageRef.RefName]
+	if len(versions) == 0 {
+		return nil, fmt.Errorf("no package versions for the package %q", pkgInstall.Spec.PackageRef.RefName)
+	}
+
+	// Carvel uses base64-encoded SVG data for IconSVGBase64, whereas we need
+	// a url, so convert to a data-url.
+	iconUrl := ""
+	if pkgMetadata.Spec.IconSVGBase64 != "" {
+		iconUrl = fmt.Sprintf("data:image/svg+xml;base64,%s", pkgMetadata.Spec.IconSVGBase64)
+	}
+
+	installedPackageSummary := &corev1.InstalledPackageSummary{
+		CurrentVersion: &corev1.PackageAppVersion{
+			PkgVersion: pkgInstall.Status.LastAttemptedVersion,
+		},
+		IconUrl: iconUrl,
+		InstalledPackageRef: &corev1.InstalledPackageReference{
+			Context: &corev1.Context{
+				Namespace: pkgMetadata.Namespace,
+				Cluster:   cluster,
+			},
+			Plugin:     &pluginDetail,
+			Identifier: pkgInstall.Name,
+		},
+		LatestMatchingVersion: &corev1.PackageAppVersion{
+			PkgVersion: versions[0].version.String(),
```

Comment:
I don't think this is necessarily true, is it? (Maybe leave at least a comment). I mean, if the `PackageInstall` has a `versionSelection.constraint` of `2.0.0`, and versions has a 3.0.0, 2.1.0 and 2.0.0, then the latest matching version is still 2.0.0, isn't it?

Reply:
Yes, absolutely; I added a temporary value for populating the field, but just forgot to add a comment explaining it :sweat_smile: 
Adding a TODO for now.